### DISCUSSION TITLE
Select item does not change when @change is used in external Alpine.data

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3563] [7. Bugs (archived)]
When using `@change` within the html tag, everything works. When using the `@change` in the external `Alpine.data`, the select value can't be changed anymore: https://codepen.io/simonferndriger/pen/QWZrLrX

**Comment by ekwoka** [comment]
Works fine for me.

Just take the `@change` off the element.

**Comment by simonferndriger** [comment]
Now you can't select "two", it resets automatically to "one". Try yourself:
https://codepen.io/simonferndriger/pen/QWZrLrX

Am Mi., 10. Mai 2023 um 12:52 Uhr schrieb Eric Kwoka <
***@***.***>:

**Reply by ekwoka** [reply]
Yeah, because you're removing the two option.

Why are you removing the two option?

Just leave it there.

**Reply by simonferndriger** [reply]
I remove only the items, that are NOT selected, because I want the dropdown icon to be right after the value, otherwise, the select element takes up the space from the longest item, even though it is not visible (at least in Windows 10). However, when I do the same with the `@change` in the html element, it works. Shouldn't the bevaiour be the same in both ways?

**Reply by simonferndriger** [reply]
Like this, it works: https://codepen.io/simonferndriger/pen/QWZrLrX

**Reply by ekwoka** [reply]
Well, I'd that's really something you need, you can just wrap it in `nextTick`

Or let the user choose when to move their focus themselves.

**Reply by simonferndriger** [reply]
Thanks. With `nextTick` it works. So, the execution order is different for the same event listener inline VS external?

**Reply by ekwoka** [reply]
Well, event listeners on the same element fire in the order they are added. I guess x-bind runs before x-model which does make sense.

**Reply by simonferndriger** [reply]
If you put it like this, it makes a lot of sense 😅 Thanks!
